src/v/iceberg/avro_decimal.h

```diff
@@ -27,9 +27,12 @@ inline bytes encode_avro_decimal(absl::int128 decimal) {
 
     for (int i = 0; i < 8; i++) {
         // NOLINTNEXTLINE(hicpp-signed-bitwise)
-        decimal_bytes[i] = (low_half >> (i * 8)) & 0xFF;
+        decimal_bytes[i] = (high_half >> ((7 - i) * 8)) & 0xFF;
         // NOLINTNEXTLINE(hicpp-signed-bitwise)
-        decimal_bytes[i + 8] = (high_half >> (i * 8)) & 0xFF;
+        decimal_bytes[i + 8] = (low_half >> ((7 - i) * 8)) & 0xFF;
+        // convert to big endian
+        decimal_bytes[i] = ss::cpu_to_be(decimal_bytes[i]);
+        decimal_bytes[i + 8] = ss::cpu_to_be(decimal_bytes[i + 8]);
```

Comment:
This does nothing right? This is a single byte so there is no endianess

Reply:
oh, now i know what you meant, i always thought that bits in a single byte are also reversed but they are not

Reply:
I wish - I think that would be simpler